server/src/main/java/org/opensearch/common/util/concurrent/ThreadContextStatePropagator.java

```diff
@@ -22,15 +22,39 @@
 public interface ThreadContextStatePropagator {
     /**
      * Returns the list of transient headers that needs to be propagated from current context to new thread context.
-     * @param source current context transient headers
+     *
+     * @param source          current context transient headers
      * @return the list of transient headers that needs to be propagated from current context to new thread context
      */
     Map<String, Object> transients(Map<String, Object> source);
```

Comment:
Could you please mark this method as @Deprecated(since = "2.12.0", forRemoval = true)`, same for `headers`, thank you